File: KruskalAlgorithm.cpp
#include<iostream>
#include<algorithm>
using namespace std;
class Edge{
public:
	int source;
	int destination;
	int weight; 
};
bool compare(Edge e1,Edge e2){
   return e1.weight<e2.weight;
}

int findParent(int v,int *parent){
	if(parent[v]==v){
		return v;
	}

	return findParent(parent[v],parent);
}
void kruskal(Edge *input,int n,int e){
	sort(input,input+e,compare);
    Edge *output=new Edge[n-1];
    int *parent=new int[n];
    for(int i=0;i<n;i++){
    	parent[i]=i;
    }

    int count=0;
	int i=0;
	while(count!=n-1){
       Edge currentEdge=input[i];

       int sourceParent=findParent(currentEdge.source,parent);
       int destinationParent=findParent(currentEdge.destination,parent);
       if(sourceParent!=destinationParent){
          output[count]=currentEdge;
          count++;  
          parent[sourceParent]=destinationParent;
       } 	   

       i++;

    }

    for(int i=0;i<n-1;i++){
    	if(output[i].source<output[i].destination){
    	  cout<<output[i].source<<" "<<output[i].destination<<" "<<output[i].weight<<endl;	
    	}else{
    		cout<<output[i].destination<<" "<<output[i].source<<" "<<output[i].weight<<endl;
    	}
    	
    }
}
int main(){
	int n,E;
	cin>>n;
	cin>>E;
    
    Edge *input=new Edge[E];

    for(int i=0;i<E;i++){
       int s,d,w;
       cin>>s;
       cin>>d;
       cin>>w; 
       input[i].source=s;
       input[i].destination=d;
       input[i].weight=w;      
    }

    kruskal(input,n,E);

}

File: KruskalAlgorithm.java
import java.util.Scanner;
  import java.util.Arrays;
  class Edge implements Comparable<Edge>{

    int v1;
    int v2; 
    int weight;
      Edge(int v1,int v2,int weight){
      this.v1=v1;
      this.v2=v2;
      this.weight=weight;
    } 

     @Override
    public int compareTo(Edge o){
      return this.weight-o.weight;
     }
  } 
  public class Solution {
   
       public static int findParent(int currentEdge,int parent[]){

        if(currentEdge==parent[currentEdge]){
          return currentEdge;
        }

        return findParent(parent[currentEdge],parent);
      }

      public static Edge[] KruskalAlgotithm(Edge edges[],int n){
          Arrays.sort(edges);

          Edge mst[]=new Edge[n-1];
           
           int parent[]=new int[n];
           for(int i=0;i<n;i++){
                parent[i]=i;
           }
          int count=0,i=0;
          while(count!=n-1){
              
              Edge currentEdge=edges[i++];
              int v1Parent=findParent(currentEdge.v1,parent);
              int v2Parent=findParent(currentEdge.v2,parent);

              if(v1Parent!=v2Parent){
                  mst[count]=currentEdge;
                  count++;
                  parent[v1Parent]=v2Parent;  
              }
            //count++;
          }
          
          return mst;

      }
    public static void main(String[] args) {
      Scanner sc = new Scanner(System.in);


      int n=sc.nextInt();
        int E=sc.nextInt();

        Edge edges[]=new Edge[E];
        
        for(int i=0;i<E;i++){
          int v1=sc.nextInt();
          int v2=sc.nextInt();
          int weight=sc.nextInt();
          Edge edge=new Edge(v1,v2,weight);
          edges[i]=edge;
        }
      
        Edge mst[]=KruskalAlgotithm(edges,n);
      
        for(int i=0;i<mst.length;i++){
            if(mst[i].v1<mst[i].v2){
              System.out.println(mst[i].v1 +" " +mst[i].v2+" "+mst[i].weight);
            }
            else{
              System.out.println(mst[i].v2+" "+mst[i].v1+" "+mst[i].weight);
            }
        }
    }
  }


// 2 errors
